src/metabase/pulse/render/body.clj

```diff
@@ -758,11 +759,17 @@
         render-fn      (if (isa? (-> cols x-axis-rowfn :effective_type) :type/Temporal)
                          js-svg/timelineseries-waterfall
                          js-svg/categorical-waterfall)
+        settings       (->js-viz x-col y-col viz-settings)
```

Comment:
This could be a single thread first form rather than 3 bindings to the same name

Reply:
sgtm